toolgenie.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/env python
+from __future__ import print_function
+from collections import namedtuple
+import os, re, readline
+import xml.etree.ElementTree as ET
+
+# https://stackoverflow.com/questions/21731043/use-of-input-raw-input-in-python-2-and-3
+# Bind raw_input to input in Python 2.
+try:
+    input = raw_input
+except NameError:
+    pass
+
+class CMSSW(namedtuple('CMSSW', 'type major mid minor extra note')):
```

Comment:
is there a reason some named tuples are initialized with a space-separated string, while others are initialized with a list?

Reply:
No, not really. Both methods are allowed. I think there is some technical difference, but it's been a long time since I actually had a reason to care.